### PR TITLE
ATTACH AS [NOT] REPLICATED

### PR DESCRIPTION
**Pull request by kirillgarbar**
Basically the same idea as https://github.com/ClickHouse/ClickHouse/pull/58746, but implementation is simpler and more user friendly.
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Added a query that allows to attach MergeTree tables as ReplicatedMergeTree and vice versa.

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
*This is an automated comment for commit 75b88042ea384d54070086076d0272f78b5052a3 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/65401/75b88042ea384d54070086076d0272f78b5052a3/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>A Sync</td><td>If it fails, ask a maintainer for help</td><td>⏳ pending</td></tr>
<tbody>
</table>
<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>Checks correctness of the PR's body</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

File: src/Databases/DatabaseOrdinary.cpp [hunk 1]
```diff
@@ -158,7 +177,8 @@ void DatabaseOrdinary::convertMergeTreeToReplicatedIfNeeded(ASTPtr ast, const Qu
 
     LOG_INFO(log, "Found {} flag for table {}. Will try to change it's engine in metadata to replicated.", CONVERT_TO_REPLICATED_FLAG_NAME, backQuote(qualified_name.getFullName()));
 
-    setReplicatedEngine(create_query, getContext());
+    checkReplicaPathExists(create_query, getContext());
+    setReplicatedMergeTreeEngine(create_query, getContext(), true);
```
Comment (tavplubix) [inline src/Databases/DatabaseOrdinary.cpp:181]:
```suggestion
    setReplicatedMergeTreeEngine(create_query, getContext(), /*replicated*/ true);
```

File: src/Interpreters/InterpreterCreateQuery.cpp [hunk 2]
```diff
@@ -1780,6 +1785,11 @@ BlockIO InterpreterCreateQuery::execute()
     bool is_create_database = create.database && !create.table;
     if (!create.cluster.empty() && !maybeRemoveOnCluster(query_ptr, getContext()))
     {
+        if (create.attach_as_replicated.has_value())
+            throw Exception(
+                ErrorCodes::SUPPORT_IS_DISABLED,
+                "ATTACH AS [NOT] REPLICATED is not supported for ON CLUSTER queries");
```
Comment (tavplubix) [inline src/Interpreters/InterpreterCreateQuery.cpp:1791]: We also should disallow queries with `attach_short_syntax = false`

**Review by tavplubix — APPROVED** [review]

**Comment by kirillgarbar** [comment]
Suppressing warning with `CLICKHOUSE_CLIENT_SERVER_LOGS_LEVEL=fatal` is not working for some reason, had to add allow_deprecated_database_ordinary to every query

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Integration tests (aarch64) [6/6]  - https://github.com/ClickHouse/ClickHouse/pull/72357

**Pull request merged**